### DISCUSSION TITLE
Elasticagent integration netflow

### DISCUSSION THREAD

**Discussion by protecme** [Security-Onion-Solutions/securityonion#11487] [2.4]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

64

### Storage for /

2to

### Storage for /nsm

20to

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hello, 

After updating SO to version 2.4.20, I'm testing the integration of modules via elacticagent.
however I have a problem with the operation
here's how I configured it: 

Firewall configuration : 
For Allow Range to push data 
<img width="1023" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/18191518-da62-4e7c-922d-1a711f66d053">
Open port 2055 TCP and UDP
<img width="1155" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/6e0ed3a0-29e4-41c3-89e0-2b5e16c6038a">
<img width="1160" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/388c62ae-091e-4e24-a7e5-b4568ceb911f">
for sensor add the customportgroup 
<img width="1159" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/85095215-7331-409b-86d9-716d8fabbeff">
<img width="1173" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/bace406e-d5b9-4558-9bd7-44b095ba5546">
<img width="1173" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/4e76dd9f-1f33-41c8-ab58-ec7332273884">

config Fleet module
<img width="1248" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/ad004a9a-4b18-42e1-8b0f-9e11172a84c0">

<img width="834" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/b2d55126-9d70-4823-99bf-c2af88b49a0d">
name space : SO  or default (is the same).

on my dashboard SO : 
<img width="1155" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/75514b46-b9e1-405b-b31e-e99565ea1af3">
Kibana dashboard netflow : 
<img width="1386" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/9e9f11b4-2d95-486e-b4ea-993c1b489f6a">

I must be doing something wrong or forgetting something, but I do have packets on the probe. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/6d1c4c8f-4cf5-45a5-9e1a-3c0cbd4c7e36)

can you help me with the integration of the module and tell me what's missing in the configuration or is it a bug in the new version? 

Many thanks

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by InfosecGoon** [comment]
Have you tried testing it with a Netflow traffic generator, in case it's something odd with the data you're sending to SO? I've had a lot of success with this one: https://github.com/nerdalert/nflow-generator

**Comment by protecme** [comment]
thank you
I tested a second firewall that works in Version 2.3 ( filebeat ) and to another collector ( supervision ) but to 2.4.20 it doesn't work.

I tested a lot of options in the fleet integration but with no result. I have the impression that the data is in elastic but is not recognized.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/194670f5-3113-49e7-836b-0dd4e5dbb0e9)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/0fcbf7ee-9983-4ca0-85ee-d04f223d55ef)


maybe a problem in the pipeline or in the module configuration, but I have exactly the same problem with another module, PANW (palo alto), and I have no data coming in.

I've consulted the official documentation but I can't find anything more.
as the netflow and panw modules are now operational since version 2.4.20, maybe there's an oversight in the commit?

if i can have a confirmation that the module works correctly and in broad outline how to set it up.

thank you

**Reply by Phillip-Samson** [reply]
AGREED, Major issues with 2.4.20 integrations I am experiencing lack of parsing fro all of my integrations
Squid
Pano
DarkTrace
Okta
None are working correctly anymore

**Reply by jaypfe** [reply]
@Bal33p  I just deployed the Okta integration, its working as it should. It picks up the correct index template from 'so-logs-okta.system'
Seems to be parsing correctly. Have you tried installing again maybe? I did not have this deployed before in any capacity.  

I am having issues with the FortiGate one, but this one seems to be hitting right on the money out of the box.

**Reply by Phillip-Samson** [reply]
So I had all my integrations working, but I had to use the Advanced feature in elastic search to link the proper templates back in 2.4.10
Since I Updated, it has been broken and shows that the template is so-logs the fix from before no longer works. it is as though it is stuck
I deleted the index templates everything and reinstalled many times. same result 
I have rolled back and tried to use the advanced feature again, but it won't take.
The datatstream labeled logs-pane.panos-default now I, I can't remember if that is normal but in the default.sls it does not have -default in their name
meanwhile, my SOME of the elastic agents are no longer sending logs in and I can't seem to figure out why as there are no errors that I can find.

elasticsearch:
  index_settings:
    so-panw-panos:
      index_sorting: False
      index_template:
        index_patterns:
          - "logs-panw.panos-*"
        template:
          settings:
            index:
              number_of_replicas: 0
        composed_of:
          - "logs-panw.panos@package"
          - "logs-panw.panos@custom"
          - "so-fleet_globals-1"
          - "so-fleet_agent_id_verification-1"
        priority: 501
        data_stream:
          hidden: false
          allow_custom_routing: false

**Reply by petergtrom** [reply]
For what it's worth, I did end up figuring this out. Especially thanks to your attached images! So thank you so much for that.

Create your hostgroup and portgroup, set your firewall. Then in Elastic Fleet, the field for "UDP host to listen on" should be the management IP address of the SO manager/sensor, whichever you're using. Once that's done, all Netflow starts populating.

If you SSH to the SO node you're attempting to send Netflow traffic to, and do a "netstat -tunlp", in the example shown, UDP port 2055 is listening on 127.0.0.1 for local host. This won't accept any ingress Netflow traffic. By changing it to the IP of the SO node, it will then listen.

In this example, 10.32.45.100 is my SO node.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/137253795/a31d6e4a-82ca-4706-bd12-598d0e14dd75)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/137253795/828e6982-16e0-4717-a36f-2456708d1880)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/137253795/3559f37b-5326-4486-bedd-685a29a8489b)

**Reply by ragebobcat80** [reply]
Did you add the so-grid-nodes_general to the Fleetserver-XXXXXXhost ?

**Reply by petergtrom** [reply]
Actually yes, I did install it to the so-grid-nodes_general.

Strangely, and this is my first experience with Fleet, I wasn't sure how to create a parent agent policy with child agent policies. It appears that if you create an Integration Policy for something like Netflow or MISP, and apply the policy to an agent, it replaces any other policy.

I found that odd....
